[INFO 18:08:30] pymarl Running command 'my_main'
[INFO 18:08:30] pymarl Started run with ID "1"
[DEBUG 18:08:30] pymarl Starting Heartbeat
[DEBUG 18:08:30] my_main Started
[INFO 18:08:30] my_main Experiment Parameters:
[INFO 18:08:30] my_main 

{   'action_selector': 'soft_policies',
    'add_value_last_step': True,
    'agent': 'rnn',
    'agent_output_type': 'pi_logits',
    'batch_size': 10,
    'batch_size_run': 10,
    'buffer_cpu_only': True,
    'buffer_size': 10,
    'checkpoint_path': '',
    'common_reward': True,
    'critic_type': 'cv_critic',
    'entropy_coef': 0.001,
    'env': 'gymma',
    'env_args': {   'key': 'marl:rware-small-4ag-v2-123',
                    'pretrained_wrapper': None,
                    'seed': 700077782,
                    'time_limit': 500},
    'epochs': 4,
    'eps_clip': 0.2,
    'evaluate': False,
    'gamma': 0.99,
    'grad_norm_clip': 10,
    'hidden_dim': 128,
    'hypergroup': None,
    'label': 'default_label',
    'learner': 'ppo_learner',
    'learner_log_interval': 500000,
    'load_step': 0,
    'local_results_path': 'results',
    'log_interval': 500000,
    'lr': 0.0005,
    'mac': 'basic_mac',
    'mask_before_softmax': True,
    'name': 'mappo',
    'obs_agent_id': True,
    'obs_individual_obs': False,
    'obs_last_action': False,
    'optim_alpha': 0.99,
    'optim_eps': 1e-05,
    'q_nstep': 10,
    'render': False,
    'repeat_id': 1,
    'reward_scalarisation': 'sum',
    'runner': 'parallel',
    'runner_log_interval': 500000,
    'save_model': True,
    'save_model_interval': 50000,
    'save_replay': False,
    'seed': 700077782,
    'standardise_returns': False,
    'standardise_rewards': False,
    't_max': 20050000,
    'target_update_interval_or_tau': 0.01,
    'test_greedy': True,
    'test_interval': 500000,
    'test_nepisode': 100,
    'use_cuda': True,
    'use_rnn': True,
    'use_tensorboard': False,
    'use_wandb': False,
    'wandb_mode': 'offline',
    'wandb_project': None,
    'wandb_save_model': False,
    'wandb_team': None}

Process Process-1:
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/runners/parallel_runner.py", line 290, in env_worker
    env = env_fn.x()
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/__init__.py", line 33, in gymma_fn
    return GymmaWrapper(**kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/gymma.py", line 39, in __init__
    self._env = gym.make(f"{key}", **kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 687, in make
    env_spec = _find_spec(id)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 531, in _find_spec
    _check_version_exists(ns, name, version)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 397, in _check_version_exists
    _check_name_exists(ns, name)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 374, in _check_name_exists
    raise error.NameNotFound(
gymnasium.error.NameNotFound: Environment `rware-small-4ag-v2-123` doesn't exist.
Process Process-2:
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/runners/parallel_runner.py", line 290, in env_worker
    env = env_fn.x()
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/__init__.py", line 33, in gymma_fn
    return GymmaWrapper(**kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/gymma.py", line 39, in __init__
    self._env = gym.make(f"{key}", **kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 687, in make
    env_spec = _find_spec(id)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 531, in _find_spec
    _check_version_exists(ns, name, version)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 397, in _check_version_exists
    _check_name_exists(ns, name)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 374, in _check_name_exists
    raise error.NameNotFound(
gymnasium.error.NameNotFound: Environment `rware-small-4ag-v2-123` doesn't exist.
Process Process-3:
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/runners/parallel_runner.py", line 290, in env_worker
    env = env_fn.x()
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/__init__.py", line 33, in gymma_fn
    return GymmaWrapper(**kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/gymma.py", line 39, in __init__
    self._env = gym.make(f"{key}", **kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 687, in make
    env_spec = _find_spec(id)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 531, in _find_spec
    _check_version_exists(ns, name, version)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 397, in _check_version_exists
    _check_name_exists(ns, name)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 374, in _check_name_exists
    raise error.NameNotFound(
gymnasium.error.NameNotFound: Environment `rware-small-4ag-v2-123` doesn't exist.
Process Process-4:
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/runners/parallel_runner.py", line 290, in env_worker
    env = env_fn.x()
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/__init__.py", line 33, in gymma_fn
    return GymmaWrapper(**kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/gymma.py", line 39, in __init__
    self._env = gym.make(f"{key}", **kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 687, in make
    env_spec = _find_spec(id)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 531, in _find_spec
    _check_version_exists(ns, name, version)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 397, in _check_version_exists
    _check_name_exists(ns, name)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 374, in _check_name_exists
    raise error.NameNotFound(
gymnasium.error.NameNotFound: Environment `rware-small-4ag-v2-123` doesn't exist.
Process Process-5:
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/runners/parallel_runner.py", line 290, in env_worker
    env = env_fn.x()
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/__init__.py", line 33, in gymma_fn
    return GymmaWrapper(**kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/gymma.py", line 39, in __init__
    self._env = gym.make(f"{key}", **kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 687, in make
    env_spec = _find_spec(id)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 531, in _find_spec
    _check_version_exists(ns, name, version)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 397, in _check_version_exists
    _check_name_exists(ns, name)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 374, in _check_name_exists
    raise error.NameNotFound(
gymnasium.error.NameNotFound: Environment `rware-small-4ag-v2-123` doesn't exist.
Process Process-6:
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/runners/parallel_runner.py", line 290, in env_worker
    env = env_fn.x()
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/__init__.py", line 33, in gymma_fn
    return GymmaWrapper(**kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/gymma.py", line 39, in __init__
    self._env = gym.make(f"{key}", **kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 687, in make
    env_spec = _find_spec(id)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 531, in _find_spec
    _check_version_exists(ns, name, version)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 397, in _check_version_exists
    _check_name_exists(ns, name)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 374, in _check_name_exists
    raise error.NameNotFound(
gymnasium.error.NameNotFound: Environment `rware-small-4ag-v2-123` doesn't exist.
Process Process-7:
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/runners/parallel_runner.py", line 290, in env_worker
    env = env_fn.x()
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/__init__.py", line 33, in gymma_fn
    return GymmaWrapper(**kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/gymma.py", line 39, in __init__
    self._env = gym.make(f"{key}", **kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 687, in make
    env_spec = _find_spec(id)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 531, in _find_spec
    _check_version_exists(ns, name, version)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 397, in _check_version_exists
    _check_name_exists(ns, name)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 374, in _check_name_exists
    raise error.NameNotFound(
gymnasium.error.NameNotFound: Environment `rware-small-4ag-v2-123` doesn't exist.
Process Process-8:
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/runners/parallel_runner.py", line 290, in env_worker
    env = env_fn.x()
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/__init__.py", line 33, in gymma_fn
    return GymmaWrapper(**kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/gymma.py", line 39, in __init__
    self._env = gym.make(f"{key}", **kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 687, in make
    env_spec = _find_spec(id)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 531, in _find_spec
    _check_version_exists(ns, name, version)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 397, in _check_version_exists
    _check_name_exists(ns, name)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 374, in _check_name_exists
    raise error.NameNotFound(
gymnasium.error.NameNotFound: Environment `rware-small-4ag-v2-123` doesn't exist.
Process Process-9:
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/runners/parallel_runner.py", line 290, in env_worker
    env = env_fn.x()
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/__init__.py", line 33, in gymma_fn
    return GymmaWrapper(**kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/gymma.py", line 39, in __init__
    self._env = gym.make(f"{key}", **kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 687, in make
    env_spec = _find_spec(id)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 531, in _find_spec
    _check_version_exists(ns, name, version)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 397, in _check_version_exists
    _check_name_exists(ns, name)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 374, in _check_name_exists
    raise error.NameNotFound(
gymnasium.error.NameNotFound: Environment `rware-small-4ag-v2-123` doesn't exist.
Process Process-10:
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/runners/parallel_runner.py", line 290, in env_worker
    env = env_fn.x()
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/__init__.py", line 33, in gymma_fn
    return GymmaWrapper(**kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/src/envs/gymma.py", line 39, in __init__
    self._env = gym.make(f"{key}", **kwargs)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 687, in make
    env_spec = _find_spec(id)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 531, in _find_spec
    _check_version_exists(ns, name, version)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 397, in _check_version_exists
    _check_name_exists(ns, name)
  File "/home/bhuv/Documents/bhuv-diffusion/epymarl/epymarl_env/lib/python3.8/site-packages/gymnasium/envs/registration.py", line 374, in _check_name_exists
    raise error.NameNotFound(
gymnasium.error.NameNotFound: Environment `rware-small-4ag-v2-123` doesn't exist.
